vumi_http_retry/workers/maintainer/tests/test_worker.py

```diff
@@ -1,5 +1,148 @@
+from twisted.internet.task import Clock
 from twisted.trial.unittest import TestCase
+from twisted.internet.defer import inlineCallbacks, returnValue, succeed
+
+from vumi_http_retry.workers.maintainer.worker import RetryMaintainerWorker
+from vumi_http_retry.retries import pending_key, ready_key, add_pending
+from vumi_http_retry.tests.redis import zitems, lvalues, delete
+
+
+class ToyRetryMaintainerWorker(RetryMaintainerWorker):
+    @inlineCallbacks
+    def setup(self, *a, **kw):
+        self.maintains = []
+        yield super(ToyRetryMaintainerWorker, self).setup(*a, **kw)
+
+    def maintain(self):
+        d = super(ToyRetryMaintainerWorker, self).maintain()
+        self.maintains.append(d)
+        return d
 
 
 class TestRetryMaintainerWorker(TestCase):
-    pass
+    @inlineCallbacks
+    def teardown_worker(self, worker):
+        yield delete(worker.redis, 'test.*')
+        yield worker.teardown()
+
+    @inlineCallbacks
+    def mk_worker(self, config):
+        config['redis_pefix'] = 'test'
+        worker = ToyRetryMaintainerWorker(config)
+
+        yield worker.setup(Clock())
+        self.addCleanup(self.teardown_worker, worker)
+
+        worker.stop_maintain_loop()
```

Comment:
Actually, an option is just the same as calling the method explicitly, will do that instead.